gamelib/Animation.py

```diff
@@ -0,0 +1,218 @@
+            self.animated_object = animated_object
+            self.refresh_screen = refresh_screen
+    
+    def start(self):
+        """Set the animation state to RUNNING.
+
+        If the animation state is not RUNNING, animation's next_frame() function return the last frame returned.
+
+        Example::
+        
+            item.animation.start()
+        """
+        self.state = RUNNING
+    
+    def pause(self):
+        """Set the animation state to PAUSED.
+
+        Example::
+        
+            item.animation.pause()
+        """
+        self.state = PAUSED
+    
+    def stop(self):
+        """Set the animation state to STOPPED.
+        
+        Example::
+        
+            item.animation.stop()
+        """
+        self.state = STOPPED
+
+    def add_frame(self,frame):
+        """Add a frame to the animation.
+
+        The frame has to be a string (that includes sprites from the Sprite module and squares from the Utils module).
+
+        Raise an exception if frame is not a string.
+
+        :param frame: The frame to add to the animation.
+        :type frame: str
+        :raise: :class:`gamelib.HacExceptions.HacInvalidTypeException`
+        
+        Example::
+
+            item.animation.add_frame(Sprite.ALIEN)
+            item.animation.add_frame(Sprite.ALIEN_MONSTER)
+        """
+        if type(frame) is not str:
+            raise HacInvalidTypeException('The "frame" parameter must be a string.')
+        self.frames.append(frame)
+    
+    def search_frame(self,frame):
+        """Search a frame in the animation.
+
+        That method is returning the index of the first occurrence of "frame".
+
+        Raise an exception if frame is not a string.
+
+        :param frame: The frame to find.
+        :type frame: str
+        :rtype: int
+        :raise: :class:`gamelib.HacExceptions.HacInvalidTypeException`
+        
+        Example::
+
+            item.animation.remove_frame( item.animation.search_frame(Sprite.ALIEN_MONSTER) )
+            
+        """
+        if type(frame) is not str:
+            raise HacInvalidTypeException('The "frame" parameter must be a string.')
+        return self.frames.index(frame)
+    
+    def remove_frame(self,index):
+        """Remove a frame from the animation.
+
+        That method remove the frame at the specified index and return it if it exists. 
+        
+        If the index is out of bound an exception is raised.
+        If the index is not an int an exception is raised.
+
+        :param index: The index of the frame to remove.
+        :type index: int
+        :rtype: str
+        :raise: IndexError, HacInvalidTypeException
+        
+        Example::
+
+            item.animation.remove_frame( item.animation.search_frame(Sprite.ALIEN_MONSTER) )
+            
+        """
+        if type(index) is not int:
+            raise HacInvalidTypeException('The "index" parameter must be an int.')
+        return self.frames.pop(index)
```

Comment:
what happens with the current index when we remove a middle frame?

Reply:
That's a very good question... Right now nothing, I'll fix that probably this way (just hot writting that I'll look into it tonight):
- If the removed frame index is greater than the current index, do nothing.
- If the removed frame is lesser or equal to the current index decrement the current index.

Reply:
Ok so that's what I implemented (algo from my previous comment).
I also added test cases to make sure to test that behavior.